src/vse_sync_pp/analyzers/ppsdpll.py

```diff
@@ -0,0 +1,209 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+import pandas as pd
+import numpy as np
+import allantools
+
+from collections import namedtuple
+from decimal import Decimal
+
+class Analyzer():
+    def __init__(self):
+        self._col1_name = 'timestamp'
+        self._col2_name = 'eecstate'
+        self._col3_name = 'phasestate'
+        self._col4_name = 'phaseoffset'
+        self._data  = pd.DataFrame(columns=['timestamp','eecstate','phasestate','phaseoffset']) 
+    def append(self, data):
+        """Append x and y data points extracted from `data`"""
+        col1_val = getattr(data, self._col1_name)
```

Comment:
you could add these to a iterable of some form say 
```python
def __init__(self):
 # ...
 self._col_names = (self._col1_name, self._col2_name, self._col3_name, self._col4_name)
```
The append func then becomes
```
self._data.loc[len(self._data)]=[getattr(data, cname) for cname in self._col_names]
```

Reply:
This could be further cleaned up using `attrgetter` 
https://docs.python.org/3/library/operator.html#operator.attrgetter
```
def __init__(self):
   self._data_getter = attrgetter(self._col1_name, self._col2_name, self._col3_name, self._col4_name)

def append(self, data):

self._data.loc[len(self._data)]=self._data_getter(data)
```

Reply:
Nice. This simplifies append method.